libs/server/ArgSlice/ScratchBufferManager.cs

```diff
@@ -29,7 +29,7 @@ internal sealed unsafe class ScratchBufferManager
         /// <summary>
         /// Current offset in scratch buffer
         /// </summary>
-        int scratchBufferOffset;
+        internal int scratchBufferOffset { get; private set; }
```

Comment:
keep this a private int field instead of a property as it is perf critical. can add a separate getter.